src/charts/grouped-bar.js

```diff
@@ -0,0 +1,969 @@
+                    let ret = {};
+
+                    values.forEach((entry) => {
+                        if (entry && entry[groupLabel]) {
+                            ret[entry[groupLabel]] = getValue(entry);
+                        }
+                    });
+                    ret.values = values; //for tooltip
+
+                    return ret;
+                })
+                .entries(data)
+                .map(function(data){
+                    return assign({}, {
+                        total:d3Array.sum( d3Array.permute(data.value, groups) ),
+                        key:data.key
+                    }, data.value);
+                });
+        }
+
+        /**
+         * Adds events to the container group if the environment is not mobile
+         * Adding: mouseover, mouseout and mousemove
+         */
+        function addMouseEvents() {
+            svg
+                .on('mouseover', handleMouseOver)
+                .on('mouseout', handleMouseOut)
+                .on('mousemove', handleMouseMove);
+        }
+
+        /**
+         * Creates the d3 x and y axis, setting orientations
+         * @private
+         */
+        function buildAxis() {
+            if (!horizontal) {
+                xAxis = d3Axis.axisBottom(xScale)
+                yAxis = d3Axis.axisLeft(yScale)
+                    .ticks(numOfVerticalTicks, valueLabelFormat)
+            } else {
+                xAxis = d3Axis.axisBottom(xScale)
+                .ticks(numOfHorizontalTicks, valueLabelFormat);
+                yAxis = d3Axis.axisLeft(yScale)
+            }
+        }
+
+        /**
+         * Builds containers for the chart, the axis and a wrapper for all of them
+         * NOTE: The order of drawing of this group elements is really important,
+         * as everything else will be drawn on top of them
+         * @private
+         */
+        function buildContainerGroups(){
+            let container = svg
+                .append('g')
+                .classed('container-group', true)
+                .attr('transform', `translate(${margin.left},${margin.top})`);
+
+            container
+                .append('g').classed('x-axis-group', true)
+                .append('g').classed('x axis', true);
+            container.selectAll('.x-axis-group')
+                .append('g').classed('month-axis', true);
+            container
+                .append('g').classed('y-axis-group axis', true);
+            container
+                .append('g').classed('grid-lines-group', true);
+            container
+                .append('g').classed('chart-group', true);
+            container
+                .append('g').classed('metadata-group', true);
+        }
+
+        /**
+         * Builds the stacked layers layout
+         * @return {D3Layout} Layout for drawing the chart
+         * @private
+         */
+        function buildLayers(){
+            let dataInitial = transformedData.map((item) => {
```

Comment:
We could skip the creation of the temporal variable 'dataInitial' and assign the result directly into 'layers'